### DISCUSSION TITLE
7490 freetz aktuell: mod_net.log

### DISCUSSION THREAD

**Discussion by harryboo** [Freetz-NG/freetz-ng#1059] [Allgemein]
Habe die aktuelle Version ausgecheckt, kompiliert und auf die Box gespielt.
Im 'mod_net.log' steht jetzt auf allen 7490er folgender Eintrag:

1970-01-01 01:00:47 -- Wrapper for multid is executing: start
1970-01-01 01:01:00 -- Starting Dnsmasq ... **_chown: unknown user nobody.nobody_**
done.
1970-01-01 01:01:05 -- Wrapper for dsld is executing: start

dnsmasq funktioniert trotzdem...

Nur bei mir so oder auch bei Anderen ?

**Comment by fda77** [comment]
https://github.com/Freetz-NG/freetz-ng/blob/master/make/pkgs/dnsmasq/files/root/etc/init.d/rc.dnsmasq#L101
in dem bereich gabs auch lang keine änderungen

**Comment by harryboo** [comment]
Ja deswegen wundert mich das gerade, weil es bei allen Boxen auftritt.
Ich habe auch 'onlinechanged' mit gebaut und in diesem log sehe ich auf 2 von 3 Boxen diesen Eintrag:

1970-01-01 01:02:39 [5591-online] sleeping
**_1970-01-01 01:02:40 [5648-onlineipv6] rejected_**
1970-01-01 01:02:56 [5591-online] approved
1970-01-01 01:02:56 [5591-online] executing /etc/onlinechanged/00-get_ip
1970-01-01 01:02:56 [5591-online] executing /etc/onlinechanged/knock
1970-01-01 01:02:58 [5591-online]  * Stopping knock ... done.
1970-01-01 01:02:58 [5591-online]  * Starting knock ... done.
1970-01-01 01:02:58 [5591-online] executing /tmp/flash/onlinechanged/onlinechanged-cgi
1970-01-01 01:02:58 [5591-online] executing /etc/onlinechanged/webdav_net
1970-01-01 01:02:58 [5591-online]  * webdav_net online status_onlinev4
1970-01-01 01:02:59 [5591-online] finished

Hatte ich auch so noch nicht.

Kann es sein das hier etwas zu schnell abgearbeitet wird ?

**Comment by fda77** [comment]
schau doch lieber mal wo der nobody abgeblieben ist. nach manuelem "rc.dnsmasq load" müsste der auf jeden fall da sein.
es wird aber da nicht gespeichert, da benutzer eh immer angelegt werden falls nicht da.
und bei viele neuen paketen es viele "modsave" sein könnten

Evtl wird vom "wrapper" kein nobody angelegt, weil das sehr früh sein kann

**Comment by harryboo** [comment]
Was noch geaendert wurde ist ja die busybox auf version 1.37.
hier gibt es den commit:
'eba9b33b4595fbb5f2a64148b1ff3daeab9b3813' - chown: stop accepting deprecated USER.GROUP syntax, only : separator is allowed

Im log steht user und group mit einem PUNKT getrennt.

Kann das damit zusammenhaengen ?

**Comment by fda77** [comment]
Das kann gut sein!
Offenbar sind neuerdings Benutzernamen mit einem "." in linux erlaubt...
https://git.busybox.net/busybox/commit/?id=eba9b33b4595fbb5f2a64148b1ff3daeab9b3813
man müsste alle "chmod" anpassen

**Comment by harryboo** [comment]
Hab mal auf die Schnelle suchen lassen:

  C:\Users\Jens\Downloads\freetz-ng-master\make\pkgs\dnsmasq\files\root\etc\default.dnsmasq\dnsmasq_conf (1 hit)
	Line 68: 		[ ! -e /tmp/.dnsmasq.timestamp ] && cp -p /etc/version /tmp/.dnsmasq.timestamp && chown nobody.nobody /tmp/.dnsmasq.timestamp
  C:\Users\Jens\Downloads\freetz-ng-master\make\pkgs\nfsd-cgi\files\root\etc\init.d\rc.nfsd (1 hit)
	Line 39: 		chown -R nobody.nobody /var/run/rpcbind
  C:\Users\Jens\Downloads\freetz-ng-master\make\pkgs\unbound\files\root\etc\init.d\rc.unbound (1 hit)
	Line 65: 		chown -R nobody.nobody /tmp/flash/$DAEMON

Ist aber heute etwas zu frueh, ich schau heute Vormittag mal....

**Comment by fda77** [comment]
hab mehr. die meisten nutzen schon den ":"
```


./make/pkgs/dnsmasq/files/root/etc/default.dnsmasq/dnsmasq_conf:                [ ! -e /tmp/.dnsmasq.timestamp ] && cp -p /etc/version /tmp/.dnsmasq.timestamp && chown nobody.nobody /tmp/.dnsmasq.timestamp

./make/pkgs/mysql/files/root/etc/init.d/rc.mysql:       chown $DAEMON.$DAEMON $MYSQL_DIR -R
./make/pkgs/mysql/files/root/etc/init.d/rc.mysql:       chown $DAEMON.$DAEMON $PID_FILE
./make/pkgs/nfsd-cgi/files/root/etc/init.d/rc.nfsd:             chown -R nobody.nobody /var/run/rpcbind

./make/pkgs/unbound/files/root/etc/init.d/rc.unbound:           chown -R nobody.nobody /tmp/flash/$DAEMON

```


* ee741fef6 use colon for chmod

**Comment by fda77** [comment]
onlinechange rejected ist wenn der event kommt bevor das laden von freetz abgeschlossen ist und verhindert nicht mehrfach ausgelöst zu werden.  also keine problem
evtl bekomst du das sonst nicht weil ipv6 aus war?


https://github.com/Freetz-NG/freetz-ng/blob/master/make/pkgs/mod/files/root/bin/onlinechanged.sh#L36

**Comment by fda77** [comment]
busybox: das alte chown einfach so abzuschalten ist schon dumm.
noch dümmer ist es die eigene doku vorab nicht zu aktualisieren:

```
./make/busybox/Config.in.busybox.1_27:          (chown 0.0 /bin/busybox; chmod 4755 /bin/busybox)
./make/busybox/Config.in.busybox.1_27:          (chown 0.0 /etc/busybox.conf; chmod 600 /etc/busybox.conf)
./make/busybox/Config.in.busybox.1_27:    chown is used to change the user and/or group ownership
./make/busybox/Config.in.busybox.1_27:          chown root.root /bin/busybox

./make/busybox/Config.in.busybox.1_37:          (chown 0.0 /bin/busybox; chmod 4755 /bin/busybox)
./make/busybox/Config.in.busybox.1_37:          (chown 0.0 /etc/busybox.conf; chmod 600 /etc/busybox.conf)
./make/busybox/Config.in.busybox.1_37:  chown is used to change the user and/or group ownership
./make/busybox/Config.in.busybox.1_37:          chown root.root /bin/busybox
```

bzw hier nach "chown " suchen https://git.busybox.net/busybox/tree/Config.in

chown aus den coreutils macht das so:
```
chown: warning: '.' should be ':': „nobody.nobody“
```

Wäre bei busybox auch den toll idee gewesen

**Comment by harryboo** [comment]
Danke das Du das mit dem Doppelpunkt schon erledigt hast....

Bin noch auf der Suche wegen dem rejected bei 'onlinechanged'... ipv6 war immer an..
War beim start bisher waiting...

Aber da wurde seit 3 Monaten nix dran gemacht. Mein letzter build war am 26.09. und mit dem geht es noch...

@fda77 : Hast Du noch eine Idee ?

**Comment by fda77** [comment]
FOS blieb gleich? Für die 7490 gibts wohl nichts mehr...
Ansonsten kann das kann alles sein. Timing, zb durch mehr oder weniger Packages, oder Scripts von dir mit "sleep"s drin.
Bei einem bekannten am tv-kabel wurde automatisch ipv6 aktiviert.
Ansonsten hab ich dazu keine aktuelle Erfahrung, da ich Fritzboxen nur noch als Modem nutze und es so kein onlinechanged gibt
